osquery/tables/system/windows/drivers.cpp

```diff
@@ -159,42 +168,63 @@ std::string getDriverImagePath(const std::string& service_key) {
     return "";
   }
 
-  // Unify the image path as systemRoot can contain systemroot and/or system32
-  std::transform(path.begin(), path.end(), path.begin(), ::tolower);
+  return kNormalizeImage(path);
+}
 
-  char systemRoot[MAX_PATH] = {0};
-  GetSystemDirectory(systemRoot, MAX_PATH);
+Status genServiceKeyMap(
+    std::map<std::string, std::string>& services_image_map) {
+  // Attempt to get all of the services image paths in the default location
+  SQL sql("SELECT key, data FROM registry WHERE path LIKE'" + kServiceKeyPath +
+          "%\\ImagePath'");
 
-  return systemRoot +
-         boost::regex_replace(path, boost::regex("^.*?system32"), "");
+  // Something went wrong
+  if (sql.rows().empty()) {
+    return Status::failure("Failed to retrieve services image path cache");
+  }
+
+  char sys_root[MAX_PATH] = {0};
+  GetSystemDirectory(sys_root, MAX_PATH);
```

Comment:
@muffins  I just quickly glanced through this https://docs.microsoft.com/en-us/windows/desktop/fileio/naming-a-file (important things start from "These are the directory management functions that no longer have MAX_PATH restrictions" )

It looks path can still be larger then MAX_PATH, if you also confirm that it's so, I think we would need to check GetSystemDirectory for error ( https://msdn.microsoft.com/en-us/library/windows/desktop/ms724373(v=vs.85).aspx )

Reply:
Good call. I pulled this logic into the `inline` function I have above, which should've been standardized before, also I don't think this was being used in this function at all, so nice catch on my unused variables ;)